### PR TITLE
MODLD-461: Change identifier tag for LCCN mapping

### PR DESCRIPTION
**Pull request by AndreiBordak**

### CODE REVIEW

File: src/main/java/org/folio/marc4ld/service/marc2ld/authority/control/AuthorityIdentifierProcessorImpl.java [hunk 1]
```diff
@@ -34,7 +35,7 @@ public void setIdentifier(Resource resource, ControlField controlField) {
     if (ObjectUtils.notEqual(controlField.getTag(), IDENTIFIER_TAG)) {
       return;
     }
-    var data = controlField.getData();
+    var data = StringUtils.deleteWhitespace(controlField.getData());
```
Comment (pkjacob) [inline src/main/java/org/folio/marc4ld/service/marc2ld/authority/control/AuthorityIdentifierProcessorImpl.java:38]: "001" is a ControlField, where as "010" is a DataField.

So, we have to change the logic here as follows
```
public void setIdentifier(Resource resource, DataField dataField) {
    if (ObjectUtils.notEqual(dataField.getTag(), IDENTIFIER_TAG)) {
      return;
    }
    var lccn = dataField.getSubfield('a')
      .getData();
   // ....

}
```

File: src/test/resources/authority/100/marc_100_concept.jsonl [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "fields" : [ {
-    "001":"sh85121033"
+    "001":"sh85121033",
+    "010":"010 field value"
```
Comment (pkjacob) [inline src/test/resources/authority/100/marc_100_concept.jsonl:4]: Let us change the data like this -
```
"001": "123456"
"010" : {
      "subfields" : [ {
        "a" : "sh 85121033"
      }
}
```

File: src/test/resources/authority/100/marc_100_concept_Ind1_equals3.jsonl [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "fields" : [ {
-    "001":"sh85121033"
+    "001":"sh85121033",
+    "010":"010 field value"
```
Comment (pkjacob) [inline src/test/resources/authority/100/marc_100_concept_Ind1_equals3.jsonl:4]: same comment here. "010" is not a control field. It is a data field.

File: src/main/java/org/folio/marc4ld/service/marc2ld/authority/control/AuthorityIdentifierProcessorImpl.java (hunk 1 above)
Comment (AndreiBordak) [reply src/main/java/org/folio/marc4ld/service/marc2ld/authority/control/AuthorityIdentifierProcessorImpl.java]: Done.

File: src/test/resources/authority/100/marc_100_concept.jsonl (hunk 2 above)
Comment (AndreiBordak) [reply src/test/resources/authority/100/marc_100_concept.jsonl]: done.

File: src/test/resources/authority/100/marc_100_concept_Ind1_equals3.jsonl (hunk 3 above)
Comment (AndreiBordak) [reply src/test/resources/authority/100/marc_100_concept_Ind1_equals3.jsonl]: done.

**Review by askhat-abishev — APPROVED** [review]

**Comment by pkjacob** [comment]
Hi @AndreiBordak, Let us add a test where "010" do not contain "$a" subfield.

ie, let us say "010" contains the following

    "010": {
      "subfields" : [ {
        "z" : "010 field value"
      } ]
     }

In this case, LCCN will be empty.

**Comment by AndreiBordak** [comment]
@pkjacob Done.

**Review by pkjacob — APPROVED** [review]

**Pull request merged**